src/librustc_incremental/assert_dep_graph.rs

```diff
@@ -59,10 +59,18 @@ use std::fs::File;
 use std::io::Write;
 use syntax::ast;
 use syntax_pos::Span;
+use rustc::ty::ProfileQueriesMsg;
 
 pub fn assert_dep_graph<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>) {
     let _ignore = tcx.dep_graph.in_ignore();
 
+    if tcx.sess.profile_queries() {
```

Comment:
Hmm, this seems like a sort of random place for this. I think i'd prefer to put it in `librustc_driver/driver.rs`, since that's the code that *started* the thread.